source/aws_s3/s3_test.go

```diff
@@ -30,17 +30,50 @@ func Test(t *testing.T) {
 			"prod/migrations/0-random-stuff/whatever.txt": "",
 		},
 	}
-	driver := s3Driver{
-		bucket:     "some-bucket",
-		prefix:     "prod/migrations/",
-		migrations: source.NewMigrations(),
-		s3client:   &s3Client,
-	}
-	err := driver.loadMigrations()
+	driver, err := WithInstance(&s3Client, &Config{
+		Bucket: "some-bucket",
+		Prefix: "prod/migrations/",
+	})
 	if err != nil {
 		t.Fatal(err)
 	}
-	st.Test(t, &driver)
+	st.Test(t, driver)
+}
+
+func TestParseURI(t *testing.T) {
```

Comment:
Simplify this to use [table driven tests](https://github.com/golang/go/wiki/TableDrivenTests)

Reply:
Done